department-of-veterans-affairs/caseflow
# Sean C/APPEALS-9097

Resolves [APPEALS-9097](https://vajira.max.gov/browse/APPEALS-9097)

### Description
Added a new modal type for the task action: Documents ready for BVA Intake Review

### Acceptance Criteria
- [x] A VHA CAMO user can select "Documents ready for Board Intake review"
- [x] A modal appear with the header "Ready for review"
- [x] The modal has the text: "This appeal will be sent to Board intake for review.\n\nPlease select where the documents for this appeal are stored."
- [x] The modal has the options: VBMS, Centralized Mail Portal, Other
- [x] The modal also has an additional optional text box for additional details
- [x] If "Other" is selected an additional text box appears and must have something entered into it for the "Send" button to be enabled
- [x] If "VBMS" or "Centralized Mail Portal" is selected, the "Send" button is enabled
- [x] The modal has a "Cancel" button that exits the user out of the modal
- [x] When the action is executed, the following occurs:
- [x] The appeal appears on the Completed tab of the VHA CAMO queue
- [x] The appeal appears on the Ready for Review tab of the BVA Intake organization queue
- [x] The User is returned to the VHA CAMO organization queue's Completed tab
- [x] A message appears confirming the case has been sent to Board Intake for review:
"You have successfully sent <appellant's name>'s case to Board Intake for review"

### Testing Plan
1. If you need to generate some cases run this command in the rails console: `10.times.map { FactoryBot.create(:vha_document_search_task) } `
2. Logged in as a VHA CAMO user go to "Your Queue"
3. From the "Switch views" dropdown, select "VHA CAMO team cases" 
4. Click one of the appeals
5. From the Actions dropdown, select "Documents ready for Board Intake review"
6. Select one of the options
7. If the selected option was "Other", type something into the text box that appears below it
8. Click the "Send" button
9. Go back to the case details screen, and scroll down to the Case Timeline section
10. Click the "View task instructions" link

### User Facing Changes

 BEFORE|AFTER
 ---|---
<img width="1481" alt="9097-Before" src="https://user-images.githubusercontent.com/110493538/222273539-3aee87b0-0379-4bae-93ad-4689bb512936.png">|<img width="1479" alt="9097-After" src="https://user-images.githubusercontent.com/110493538/222273562-d81d5bb5-4132-4728-b3a8-a7354cf1356f.png">

<img width="503" alt="Screen Shot 2023-03-01 at 4 05 36 PM" src="https://user-images.githubusercontent.com/110493538/222276290-26538fbb-0944-473f-b191-44479c64d27a.png">
<img width="499" alt="Screen Shot 2023-03-01 at 4 05 47 PM" src="https://user-images.githubusercontent.com/110493538/222276305-19b50df1-9c14-48ca-a489-9a4cf6f47532.png">
<img width="526" alt="Screen Shot 2023-03-01 at 4 06 05 PM" src="https://user-images.githubusercontent.com/110493538/222276335-cba800b1-6fc5-4364-801f-ebe6ff4a77c1.png">
